src/aap_eda/api/views/user.py

```diff
@@ -156,15 +154,7 @@ def get_queryset(self):
         )
 
     def perform_create(self, serializer):
-        try:
-            serializer.save(user=self.request.user)
-        except django.db.utils.IntegrityError:
-            name_exists = models.AwxToken.objects.filter(
-                user=self.request.user, name=serializer.validated_data["name"]
-            ).exists()
-            if name_exists:
-                raise Conflict("Token with this name already exists.")
-            raise Conflict
+        serializer.save(user=self.request.user)
```

Comment:
Sanity check of diff - `perform_create` override here is still necessary because the intent is to get the user from the _request_, not from the _request data_.

Reply:
Correct. This POST endpoint doesn't accept `user` data but rather gets the user making the request to assign in the AxwToken object. This makes sense because users can only create tokens for themselves, not for other users.